### PR TITLE
add cancel button guidance and dos and dont's

### LINKED ISSUE #2128: Button Guidance - Implement Cancel button guidance on site (based on finished designs in #2119)
Add @Keeth-HPe's guidance from this [Figma file](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=4688%3A9756) to [Button page](https://design-system.hpe.design/components/button?q=butt) on site. Please include the do's and don't example.

In terms of order on page, it might make sense to add this right after "Buttons vs. Anchors", before "Toggle Buttons"

### PR DESCRIPTION
**Pull request by britt6612**
[Deploy Preview](https://deploy-preview-2140--keen-mayer-a86c8b.netlify.app/components/button?q=button#cancel-buttons)
[Figma](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=667%3A385)

#### What does this PR do?
This PR add the button guidance for cancel buttons
#### Where should the reviewer start?
button.mdx
#### What testing has been done on this PR?
design system site
In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [x] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### How should this be manually tested?
design system site
#### What are the relevant issues?
closes #2128 
#### Should this PR be mentioned in Design System updates?
sure
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      <Heading level={2} margin="none">
+        Rule Ordering
+      </Heading>
+      <Text>
+        This rule set is for manipulating ASM/ARM field in departments and
+        products. Also changing units to add chargeback information
+      </Text>
+    </Header>
+    <Box gap="small" align="center" justify="center">
+      <Spinner size="medium" />
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:21]: Why are we rendering a spinner here as opposed to just showing some demo set of rules? I find the spinner distracting/confusing.

File: aries-site/src/pages/components/button.mdx [hunk 2]
```diff
@@ -62,6 +64,26 @@ The HTML elements for buttons and links (a.k.a. anchors) describe a very specifi
 - Use buttons when you are performing an action, such as: "submit," "merge," "create new," "upload," etc.
 - An action almost always occurs on the same page
 
+### Cancel Buttons
+
+Cancel buttons enable users to exit out of or dismiss the current screen in order to go back to the previous one. Having an explicit way to cancel provides a fallback to safety as it helps discard any unwanted changes. 
+
+#### Styling
+Visually emphasizing the cancel button should be avoided to minimize user confusion. The cancel button should appear neutral to signify the non-committal action. For this reason, cancel buttons should be presented in the Default Button styling.
```
Comment (taysea) [inline aries-site/src/pages/components/button.mdx:72]:
```suggestion
Cancel buttons should be presented in the default button styling. The neutral appearance signifies the non-committal action. Visually emphasizing the cancel button should be avoided to minimize user confusion.
```
Leading with the action we want people to take (use default button) and following it up with the reasoning.

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js (hunk 1 above)
Comment (britt6612) [reply aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:21]: Keeth is going to update designs so I will fix with her updates

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+export const ButtonGoodCancelPreview = () => (
+  <Box width="large" gap="medium" pad="small">
+    <Header
+      direction="column"
+      align="start"
+      gap="xxsmall"
+      pad={{ horizontal: 'xxsmall' }}
+    >
+      <Heading level={2} margin="none">
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:12]:
```suggestion
      <Heading level={2} margin="none" size="small">
```
In most designs from teams, I've seen use of size small.

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        Delete
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:13]:
```suggestion
        Delete item?
```
From work done by Ephrem, there is recommendation to phrase the heading as a question that is specific about what will happen.

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      </Heading>
+      <Text size="xlarge">
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:15]:
```suggestion
      <Text size="large">
```
Typically, I've seen this as size large.

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        Deleting this item? Consider previewing results to see the effects of
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:16]:
```suggestion
        Deleting this item cannot be undone. Consider previewing results to see the effects of deleting the item.
```

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        deleting the iteam.
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:17]:
```suggestion
```

Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:17]:
```suggestion
        deleting the item.
```

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js (hunk 3 above)
Comment (britt6612) [reply aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:12]: sounds good

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+      </Text>
+    </Header>
+    <Box direction="row" justify="end" gap="small">
+      <Button label="Cancel" onClick={() => {}} />
+      <Button secondary label="Preview Results" onClick={() => {}} />
+      <Button primary label="Save" onClick={() => {}} />
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:23]:
```suggestion
      <Button primary label="Delete Item" onClick={() => {}} />
```
Based on research by Ephrem, the primary action should clearly represent the action (in this case Delete).

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js (hunk 5 above)
Comment (britt6612) [reply aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:15]: maybe I should change to `heading` in designs she had `27px`

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Box, Button, Heading, Header, Text } from 'grommet';
+
+export const ButtonGoodCancelPreview = () => (
+  <Box width="large" gap="medium" pad="small">
+    <Header
+      direction="column"
+      align="start"
+      gap="xxsmall"
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:9]:
```suggestion
      gap="xsmall"
```

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js (hunk 4 above)
Comment (britt6612) [reply aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js]: got it thank you for that I will forward that info to keeth

**Comment by britt6612** [comment]
going to convert to draft until I touch base with Maggie and Ephrem

File: aries-site/src/examples/components/button/DosDonts/ButtonBadCancelPreview.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+      direction="column"
+      align="start"
+      gap="xsmall"
+      pad={{ horizontal: 'xxsmall' }}
+    >
+      <Heading level={2} margin="none" size="small">
+        Delete item?
+      </Heading>
+      <Text size="large">
+        Deleting this item cannot be undone. Consider previewing results to see
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonBadCancelPreview.js:16]: Now that we've removed the "Preview Results" button, let's remove the "Consider previewing ..." sentence.

File: aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+      direction="column"
+      align="start"
+      gap="xsmall"
+      pad={{ horizontal: 'xxsmall' }}
+    >
+      <Heading level={2} margin="none" size="small">
+        Delete item?
+      </Heading>
+      <Text size="large">
+        Deleting this item cannot be undone. Consider previewing results to see
```
Comment (taysea) [inline aries-site/src/examples/components/button/DosDonts/ButtonGoodCancelPreview.js:16]: Same as other comment

File: aries-site/src/pages/components/button.mdx [hunk 12]
```diff
@@ -62,6 +64,26 @@ The HTML elements for buttons and links (a.k.a. anchors) describe a very specifi
 - Use buttons when you are performing an action, such as: "submit," "merge," "create new," "upload," etc.
 - An action almost always occurs on the same page
 
+### Cancel Buttons
+
+Cancel buttons enable users to exit out of or dismiss the current screen in order to go back to the previous one. Having an explicit way to cancel provides a fallback to safety as it helps discard any unwanted changes. 
+
+#### Styling
```
Comment (taysea) [inline aries-site/src/pages/components/button.mdx:71]:
```suggestion
#### Styling

```

**Review by taysea — COMMENTED** [review]
Just two small comments.

File: aries-site/src/pages/components/button.mdx [hunk 13]
```diff
@@ -62,6 +64,27 @@ The HTML elements for buttons and links (a.k.a. anchors) describe a very specifi
...
+Cancel buttons should be presented in the default button styling. The neutral appearance signifies the non-committal action. Visually emphasizing the cancel button should be avoided to minimize user confusion.
+
+<BestPracticeGroup>
+  <Example
+    bestPractice={{type: 'do', message: 'Always use default button styling for cancel button, and keep it aligned in accordance with use case, following the hierarchy.'}}
+  >
+    <ButtonGoodCancelPreview />
+  </Example>
+  <Example
+    bestPractice={{type: 'dont', message: 'Avoid using Secondary Button styling for Cancel button.'}}
```
Comment (taysea) [inline aries-site/src/pages/components/button.mdx:82]:
```suggestion
    bestPractice={{type: 'dont', message: 'Don't use Secondary Button styling for Cancel button.'}}
```

File: aries-site/src/pages/components/button.mdx [hunk 14]
```diff
@@ -62,6 +64,27 @@ The HTML elements for buttons and links (a.k.a. anchors) describe a very specifi
 - Use buttons when you are performing an action, such as: "submit," "merge," "create new," "upload," etc.
 - An action almost always occurs on the same page
 
+### Cancel Buttons
+
+Cancel buttons enable users to exit out of or dismiss the current screen in order to go back to the previous one. Having an explicit way to cancel provides a fallback to safety as it helps discard any unwanted changes. 
+
+#### Styling
+
+Cancel buttons should be presented in the default button styling. The neutral appearance signifies the non-committal action. Visually emphasizing the cancel button should be avoided to minimize user confusion.
```
Comment (halocline) [inline aries-site/src/pages/components/button.mdx:73]: 👍

**Review by halocline — APPROVED** [review]

**Pull request merged**